bug1381516.kp/knowledge.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Bug 1381516 - How Bad Is Bug 1380880?
+authors:
+- chutten
+tags:
+- investigation
+- keyed histograms
+- archaeology
+created_at: 2017-07-17 00:00:00
+updated_at: 2017-07-17 16:05:47.117913
+tldr: How broadly and how deeply do the effects of bug 1380880 extend?
+---
+### How many keyed histograms have identical keys across processes?
+
+In [bug 1380880](https://bugzilla.mozilla.org/show_bug.cgi?id=1380880) :billm found that keyed histograms recorded on different processes would be aggregated together if their keys matched.
+
+How often does this happen in practice? How long has this been happening?
+
+
+```python
+import ujson as json
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as np
+import plotly.plotly as py
+
+from plotly.graph_objs import *
+from moztelemetry import get_pings_properties, get_one_ping_per_client
+from moztelemetry.dataset import Dataset
+
+%matplotlib inline
+```
+### Which keyed histograms share keys across processes?
+
+The whole child-process client aggregation thing was introduced by [bug 1218576](https://bugzilla.mozilla.org/show_bug.cgi?id=1218576) back in September of 2016 for Firefox 52. So that's the earliest this could have started.
+
+
+```python
+pings = Dataset.from_source("telemetry") \
+    .where(docType='main') \
+    .where(appVersion=lambda x: x.startswith("52")) \
+    .where(appUpdateChannel="nightly") \
+    .records(sc, sample=0.1)
```

Comment:
Seems like it belongs in the Dataset API reference and not DTMO

https://bugzilla.mozilla.org/show_bug.cgi?id=1382205